src/pages/workspaces/workspace/DeleteWorkspaceModal.js

```diff
@@ -26,12 +75,17 @@ const DeleteWorkspaceModal = ({ workspace: { workspace: { namespace, name, bucke
     }
   }
 
+  const isDeleteDisabledFromApps = () => hasApps() && !_.isEmpty(getNonDeletableApps())
+
   return h(Modal, {
     title: 'Delete workspace',
     onDismiss,
     okButton: h(ButtonPrimary, {
-      disabled: _.toLower(deleteConfirmation) !== 'delete workspace',
-      onClick: () => deleteWorkspace()
+      disabled: _.toLower(deleteConfirmation) !== 'delete workspace' || isDeleteDisabledFromApps(),
+      onClick: () => deleteWorkspace(),
+      tooltip: _.toLower(deleteConfirmation) !== 'delete workspace' ?
+        isDeleteDisabledFromApps() ? 'You must ensure all apps in this workspace are deletable' : 'You must type the confirmation message' :
+        'Delete Workspace'
```

Comment:
Nested ternaries can be very hard to parse. Instead, consider using a `Utils.cond`:
```suggestion
      tooltip: Utils.cond(
        [isDeleteDisabledFromApps(), () => 'You must ensure all apps in this workspace are deletable'],
        [_.toLower(deleteConfirmation) !== 'delete workspace', () => 'You must type the confirmation message'],
        () => 'Permanently delete workspace'
      )
```

Note that I'm not sure the default tooltip is needed at all.